storage: TestProtectedTimestampRecordApplies failed

[(storage).TestProtectedTimestampRecordApplies failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1731611&tab=buildLog) on [master@3e176a89518d09b35fa35cf0b1873680233071df](https://github.com/cockroachdb/cockroach/commits/3e176a89518d09b35fa35cf0b1873680233071df):

```
=== RUN   TestProtectedTimestampRecordApplies
--- FAIL: TestProtectedTimestampRecordApplies (2.99s)
=== RUN   TestProtectedTimestampRecordApplies/record_does_not_exist
I200205 20:41:13.973192 12815 util/stop/stopper.go:539  quiescing
    --- FAIL: TestProtectedTimestampRecordApplies/record_does_not_exist (2.29s)
        require.go:794: 
            	Error Trace:	replica_protected_timestamp_test.go:144
            	            				replica_protected_timestamp_test.go:262
            	Error:      	Received unexpected error:
            	            	[NotLeaseHolderError] r1: replica (n1,s1):1 not lease holder; lease holder unknown
            	Test:       	TestProtectedTimestampRecordApplies/record_does_not_exist

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestProtectedTimestampRecordApplies PKG=./pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestProtectedTimestampRecordApplies.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(storage).TestProtectedTimestampRecordApplies failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1732191&tab=buildLog) on [master@51477991e358c2164b9f21c8cb46cdebb7dc4bfb](https://github.com/cockroachdb/cockroach/commits/51477991e358c2164b9f21c8cb46cdebb7dc4bfb):

```
=== RUN   TestProtectedTimestampRecordApplies
--- FAIL: TestProtectedTimestampRecordApplies (1.79s)
=== RUN   TestProtectedTimestampRecordApplies/successful_refresh_does_not_update_timestamp_(assertion_failure)
I200206 02:17:10.734716 13249 util/stop/stopper.go:539  quiescing
    --- FAIL: TestProtectedTimestampRecordApplies/successful_refresh_does_not_update_timestamp_(assertion_failure) (1.55s)
        require.go:172: 
            	Error Trace:	replica_protected_timestamp_test.go:249
            	            				replica_protected_timestamp_test.go:262
            	Error:      	Error message not equal:
            	            	expected: "cache was not updated after being refreshed"
            	            	actual  : "[NotLeaseHolderError] r1: replica (n1,s1):1 not lease holder; lease holder unknown"
            	Test:       	TestProtectedTimestampRecordApplies/successful_refresh_does_not_update_timestamp_(assertion_failure)

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestProtectedTimestampRecordApplies PKG=./pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestProtectedTimestampRecordApplies.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/storage/replica_protected_timestamp_test.go
+++ b/pkg/storage/replica_protected_timestamp_test.go
@@ -256,6 +256,9 @@ func TestProtectedTimestampRecordApplies(t *testing.T) {
 			tsc := TestStoreConfig(nil)
 			mc := &manualCache{}
 			tsc.ProtectedTimestampCache = mc
+			// Under extreme stressrace scenarios the single replica can somehow
+			// lose the lease. Make the timeout extremely long.
+			tsc.RaftConfig.RangeLeaseRaftElectionTimeoutMultiplier = 100
 			stopper := stop.NewStopper()
 			tc.StartWithStoreConfig(t, stopper, tsc)
 			stopper.Stop(ctx)

